modules/solid_mechanics/src/bcs/ExplicitDirichletBCBase.C

```diff
@@ -0,0 +1,65 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ExplicitMixedOrder.h"
+#include "ExplicitDirichletBCBase.h"
+#include "MooseError.h"
+#include "NonlinearSystemBase.h"
+#include "libmesh/numeric_vector.h"
+#include <iostream>
+#include <ostream>
+
+InputParameters
+ExplicitDirichletBCBase::validParams()
+{
+  InputParameters params = NodalBC::validParams();
+  return params;
+}
+
+ExplicitDirichletBCBase::ExplicitDirichletBCBase(const InputParameters & parameters)
+  : NodalBC(parameters),
+    _mass_diag(initMassDiag()),
+    _u_old(_var.nodalValueOld()),
+    _u_dot_old(_var.nodalValueDotOld()),
+    _exp_integrator(
+        dynamic_cast<const ExplicitMixedOrder *>(&_sys.getTimeIntegrator(_var.number())))
+{
+  if (!_exp_integrator)
+    mooseError("Time integrator for the variable is not of the right type.");
+}
+
+Real
+ExplicitDirichletBCBase::computeQpResidual()
+{
+  // Get dof for current var
+  const auto dofnum = _variable->nodalDofIndex();
+  Real resid = 0;
+  // Compute residual to enforce BC based on time order
+  switch (_var_time_order)
+  {
+    case ExplicitMixedOrder::FIRST:
+      resid = (computeQpValue() - _u_old) / _dt;
+      resid /= -_mass_diag(dofnum);
+      break;
+
+    case ExplicitMixedOrder::SECOND:
+      Real avg_dt = (_dt + _dt_old) / 2;
+      resid = (computeQpValue() - _u_old) / (avg_dt * _dt) - (_u_dot_old) / avg_dt;
+      resid /= -_mass_diag(dofnum);
+      break;
+  }
```

Comment:
How does this work if a variable appears in the equation both as a first AND a second time derivative?

Like $\ddot{u} + \dot{u} + u = 0$?

Reply:
In that case it won't satisfy the boundary condition exactly, but it shouldn't stray very far since we would be taking very small timesteps. Definitely something that can be added in the future.

Reply:
But in the future, if we want to come back and fix this, we need more information. In your implementation, we would only know the highest time derivative order associated with a variable. What we need to know instead is each and every time derivative order that is associated with a variable.

At the very minimum we should create an issue to remind us about this problem.

Reply:
agreed, I am going to make a "super" issue with a few problems I found with explicit methods.